org.scala-ide.sdt.core/src/org/scalaide/ui/internal/editor/ModelBuilder.scala

```diff
@@ -0,0 +1,257 @@
+    val b =super.<=(n)
+    n match {
+      case that:ValNode =>
+        val rt= returnType
+        returnType=that.returnType
+        b || rt != returnType
+      case _ => b
+    }
+  }
+}
+
+case class VarNode(name: String, parent: ContainerNode, rt: String)
+    extends Node(true) with HasModifiers with HasReturnType {
+  override def displayName = name
+  override def key = MethodKey(name)
+  override def <= (n:Node):Boolean={
+    val b =super.<=(n)
+    n match {
+      case that:VarNode =>
+        val rt= returnType
+        returnType=that.returnType
+        b || rt != returnType
+      case _ => b
+    }
+  }
+}
+
+
+
+object ModelBuilder extends HasLogger {
+
+  def buildTree(comp: IScalaPresentationCompiler,src:SourceFile):RootNode={
+    import comp._
+    import scala.reflect.internal.Flags._
+    def setPos(n: Node, t: Tree) = {
+      if (t.pos.isDefined) {
+        n.end = t.pos.end
+        n.start = t.pos.start
+      }
+    }
+    def updateTree(parent: ContainerNode, t: Tree): Unit = {
+      t match {
+        case Template(_, _, _) => t.children.foreach(x => updateTree(parent, x))
+        case PackageDef(pid, stats) => {
+          val ch = PackageNode(pid.toString(), parent)
+          setPos(ch, t)
+          parent.addChild(ch)
+          t.children.foreach(x => updateTree(parent, x))
+        }
+        case ClassDef(mods, name, tpars, templ) => {
+          //logger.info("ClassDef "+mods+", "+name)//+", "+tpars+", "+templ )
+          val ch = ClassNode(name.toString(), parent)
+          setPos(ch, t)
+          parent.addChild(ch)
+          ch.setFlags(mods.flags)
+          t.children.foreach(x => updateTree(ch, x))
+        }
+        case `noSelfType` =>
+        case ValDef(mods, name, tpt, rsh) => {
+          val ch =  if ((mods.flags & MUTABLE) == 0)
+              ValNode(name.toString(), parent, tpt.toString())
+            else
+              VarNode(name.toString(), parent, tpt.toString())
+          setPos(ch, t)
+          ch.setFlags(mods.flags)
+          parent.addChild(ch)
+        }
+        //        case Select(qualifier: Tree, name: Name) => logger.info("Select "+qualifier+", "+name)
+        //        case TypeDef(mods: Modifiers, name: TypeName, tparams: List[TypeDef], rhs: Tree) => logger.info("TypeDef "+name)
+        //        case AppliedTypeTree(tpt, args) =>logger.info("AppliedTypeTree "+ tpt+", "+args)
+        //        case Apply(fun, args) => logger.info("Apply "+ fun)
+        case DefDef(mods, name, tparamss, vparamss, tpt, rsh) =>
+          if ("<init>" != name.toString) {
+            def typeList = {
+              vparamss.map { x => x.map { v => v.tpt.toString() } }
+            }
+            val ch =  MethodNode(name.toString(), parent, typeList)
+            //logger.info("DefDef " + name + ", modes=" + mods + ", tpt=" + tpt.hasSymbolField + ", " + rsh)
+            ch.returnType = if (tpt.hasSymbolField) Some(tpt.toString) else None
```

Comment:
It is just a synthetic symbol (constructor) which doesn't hold any new information then a class node itself.  And name (<init>) looks weird. But I just realized that the current implementation has a filter which removes synthetic symbols. Probably I have to do the same.